packages/wallet-sdk/src/relay/WalletSDKRelayAbstract.ts

```diff
@@ -53,50 +40,51 @@ export abstract class WalletSDKRelayAbstract {
     decimals?: number,
     image?: string,
     chainId?: string
-  ): CancelablePromise<WatchAssetResponse>;
+  ): CancelablePromise<Web3Response<'watchAsset'>>;
 
   abstract selectProvider(
     providerOptions: ProviderType[]
-  ): CancelablePromise<SelectProviderResponse>;
+  ): CancelablePromise<Web3Response<'selectProvider'>>;
 
   abstract switchEthereumChain(
     chainId: string,
     address?: string
-  ): CancelablePromise<SwitchEthereumChainResponse>;
+  ): CancelablePromise<Web3Response<'switchEthereumChain'>>;
 
   abstract signEthereumMessage(
     message: Buffer,
     address: AddressString,
     addPrefix: boolean,
     typedDataJson?: string | null
-  ): CancelablePromise<SignEthereumMessageResponse>;
+  ): CancelablePromise<Web3Response<'signEthereumMessage'>>;
 
   abstract ethereumAddressFromSignedMessage(
     message: Buffer,
     signature: Buffer,
     addPrefix: boolean
-  ): CancelablePromise<EthereumAddressFromSignedMessageResponse>;
+  ): CancelablePromise<Web3Response<'ethereumAddressFromSignedMessage'>>;
 
   abstract signEthereumTransaction(
     params: EthereumTransactionParams
-  ): CancelablePromise<SignEthereumTransactionResponse>;
+  ): CancelablePromise<Web3Response<'signEthereumTransaction'>>;
 
   abstract signAndSubmitEthereumTransaction(
     params: EthereumTransactionParams
-  ): CancelablePromise<SubmitEthereumTransactionResponse>;
+  ): CancelablePromise<Web3Response<'submitEthereumTransaction'>>;
 
   abstract submitEthereumTransaction(
     signedTransaction: Buffer,
     chainId: IntNumber
-  ): CancelablePromise<SubmitEthereumTransactionResponse>;
+  ): CancelablePromise<Web3Response<'submitEthereumTransaction'>>;
 
-  abstract scanQRCode(regExp: RegExpString): CancelablePromise<ScanQRCodeResponse>;
+  abstract scanQRCode(regExp: RegExpString): CancelablePromise<Web3Response<'scanQRCode'>>;
 
-  abstract genericRequest(data: object, action: string): CancelablePromise<GenericResponse>;
+  abstract genericRequest(data: object, action: string): CancelablePromise<Web3Response<'generic'>>;
 
-  abstract sendRequest<T extends Web3Request, U extends Web3Response>(
-    request: T
-  ): CancelablePromise<U>;
+  abstract sendRequest<
+    RequestMethod extends SupportedWeb3Method,
+    ResponseMethod extends SupportedWeb3Method = RequestMethod,
+  >(request: Web3Request<RequestMethod>): CancelablePromise<Web3Response<ResponseMethod>>;
```

Comment:
it allows to take different Web3Method for request and response to support cases like signAndSubmit 🥲 

![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/671b2bb6-5b25-42ac-8145-7d79a85fa1cb)